### ISSUE TITLE
RET505 false positive

### ISSUE THREAD

**Issue by trag1c** [charliermarsh/ruff#2711]
It seems like it's not looking behind enough:
```py
def __sub__(self, other: Any) -> Array[T]:
    new_array = self.val.copy()
    if isinstance(other, Array):
        ...  # no returns here
    elif isinstance(other, Number):
        ...  # no returns here
    elif other is NULL:
        return self.val.pop()
    else:  # RET505 Unnecessary `else` after `return` statement
        raise NotDefinedError(f"Array - {get_type_name(other)}")
    return Array(new_array)
```

**Comment by charliermarsh** [comment]
The dreaded `RET505`...

**Comment by charliermarsh** [comment]
I believe this was fixed in #2881.
